controllers/kubeadmconfig_controller.go

```diff
@@ -212,6 +207,12 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 			}
 		}
 
+		// If the cluster infrastructure is not ready and the spec doesn't have a ControlPlaneEndpoint, requeue.
+		if !cluster.Status.InfrastructureReady && config.Spec.ClusterConfiguration.ControlPlaneEndpoint == "" {
```

Comment:
Is the vsphere provider setting ControlPlaneEndpoint prior to InfrastructureReady? What determines InfrastructureReady for that provider?